src/lang-promql/lsp/model.ts

```diff
@@ -0,0 +1,21 @@
+// TODO consider to use the interface coming from LSP
+// https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion
+export interface TextEdit {
```

Comment:
This package should provide the types used here: https://www.npmjs.com/package/vscode-languageserver-types

Reply:
I'm not sure yet if we should use it or not. I guess it will depend if we succeed to cover all lsp feature or not.
I don't really want to import a potentially big dependency for just one interface needed

Reply:
let's see

Reply:
After passing the Typescript compiler and maybe a Minifier, all the unnecessary Code will probably be optimized away, anyway.

And these types are quite complex. Copying them doesn't seem like a good idea to me.

Reply:
well tree shaking is more on webpack side actually. But yes I see your point here.